fault-proof/tests/e2e.rs

```diff
@@ -293,20 +287,178 @@ mod e2e {
 
         proposer_handle.abort();
 
-        for (idx, address) in mock_game_addresses.iter().enumerate() {
-            let legacy_game =
-                MockPermissionedDisputeGameView::new(*address, env.anvil.provider.clone());
-            let status_raw = legacy_game.status().call().await?;
-            let status = GameStatus::try_from(status_raw).with_context(|| {
-                format!("Failed to decode mock permissioned game status for {address}")
-            })?;
-            assert_eq!(
-                status,
-                GameStatus::IN_PROGRESS,
-                "mock permissioned game {idx} should remain untouched"
-            );
+        let legacy_game =
+            MockPermissionedDisputeGameView::new(mock_game_address, env.anvil.provider.clone());
+        let status_raw = legacy_game.status().call().await?;
+        let status = GameStatus::try_from(status_raw).with_context(|| {
+            format!("Failed to decode mock permissioned game status for {mock_game_address}")
+        })?;
+        assert_eq!(
+            status,
+            GameStatus::IN_PROGRESS,
+            "mock permissioned game should remain untouched"
+        );
+
+        Ok(())
+    }
+
+    // Ensures the proposer can handle a game type transition while running.
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_game_type_transition_while_proposer_running() -> Result<()> {
```

Comment:
After seeing this test, I think we should add a condition to the proposer to prevent game creation when the configured game type is not respected game type.

From the test logs, I see the proposer is actually trying to create games.

I also think there should be also a check before the respected game type update to the portal which checks that game count didn't increase.